integtest/spec/all_books_change_detection_spec.rb

```diff
@@ -351,6 +351,32 @@ def chapter(index)
           )
           include_examples 'second build is noop'
         end
+        context 'when --keep_hash is specified and there is a new branch' do
```

Comment:
This was my first attempt to reproduce the problem. It didn't, but it did catch some interesting cases while I was working on this issue so I want to keep it.